internal/component/loki/source/file/decompresser.go

```diff
@@ -109,12 +109,18 @@ func newDecompressor(
 		labels:            labels,
 		labelsStr:         labelsStr,
 		running:           atomic.NewBool(false),
+		stopped:           atomic.NewBool(false),
 		position:          pos,
+		posquit:           make(chan struct{}),
+		posdone:           make(chan struct{}),
 		decoder:           decoder,
 		cfg:               cfg,
 		componentStopping: componentStopping,
 	}
 
+	// updatePosition closes the channel t.posdone on exit
+	go decompressor.updatePosition()
```

Comment:
Should not start it here because it a decompressor will be recreated on on every update to the component and then we check if the task already exists. Working on a fix to that